File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/UrlSigner.java
package com.getstream.sdk.chat;

public interface UrlSigner {
    String signFileUrl(String url);

    String signImageUrl(String url);

    class DefaultUrlSigner implements UrlSigner {

        @Override
        public String signFileUrl(String url) {
            return url;
        }

        @Override
        public String signImageUrl(String url) {
            return url;
        }
    }
}

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/UrlSigner.kt
package com.getstream.sdk.chat

import java.io.Serializable

public interface UrlSigner : Serializable {
    public fun signFileUrl(url: String): String
    public fun signImageUrl(url: String): String

    public class DefaultUrlSigner : UrlSigner {
        override fun signFileUrl(url: String): String = url
        override fun signImageUrl(url: String): String = url
    }
}
